kie-server-parent/kie-server-controller/kie-server-controller-client/src/main/java/org/kie/server/controller/client/websocket/WebSocketKieServerControllerClient.java

```diff
@@ -43,31 +45,45 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(WebSocketKieServerControllerClient.class);
 
-    private WebSocketClient client;
+    private KieServerMessageHandlerWebSocketClient client;
+    private KieServerControllerNotificationWebSocketClient notificationClient;
 
     public WebSocketKieServerControllerClient(final String controllerUrl,
                                               final String userName,
                                               final String password,
-                                              final String token) {
+                                              final String token,
+                                              final EventHandler handler) {
         this(controllerUrl,
              userName,
              password,
              token,
-             new WebSocketClientImpl());
+             new KieServerMessageHandlerWebSocketClient(),
+             handler == null ? null : new KieServerControllerNotificationWebSocketClient(new WebSocketEventHandler(handler)));
     }
 
     public WebSocketKieServerControllerClient(final String controllerUrl,
                                               final String userName,
                                               final String password,
                                               final String token,
-                                              final WebSocketClient client) {
+                                              final KieServerMessageHandlerWebSocketClient client,
+                                              final KieServerControllerNotificationWebSocketClient notificationClient) {
         this.client = client;
         this.client.init(WebSocketClientConfiguration.builder()
-                                 .controllerUrl(controllerUrl)
+                                 .controllerUrl(controllerUrl + "/management")
                                  .userName(userName)
                                  .password(password)
                                  .token(token)
                                  .build());
+        if(notificationClient != null){
```

Comment:
can it actually be null or we always have at least logging notification service in place? just tingling if we can just about another condition but if it is not the case then so be it

Reply:
In here, I'm actually avoiding to open a new websocket connection(different endpoint) in case this client instance isnt interested in receiving notifications/events. So just limiting resources to what is actually needed.

Reply:
+1